internal/store/device.go

```diff
@@ -24,6 +25,11 @@ type Device interface {
 	Create(ctx context.Context, orgId uuid.UUID, device *api.Device, callback DeviceStoreCallback) (*api.Device, error)
 	Update(ctx context.Context, orgId uuid.UUID, device *api.Device, fieldsToUnset []string, fromAPI bool, callback DeviceStoreCallback) (*api.Device, error)
 	List(ctx context.Context, orgId uuid.UUID, listParams ListParams) (*api.DeviceList, error)
+	Count(ctx context.Context, orgId uuid.UUID, listParams ListParams) (int64, error)
+	UnmarkRolloutSelection(ctx context.Context, orgId uuid.UUID, fleetName string) error
+	MarkRolloutSelection(ctx context.Context, orgId uuid.UUID, listParams ListParams, limit *int) error
+	CompletionCounts(ctx context.Context, orgId uuid.UUID, owner string, templateVersion string, updateTimeout *time.Duration) ([]CompletionCount, error)
+	CountByLabels(ctx context.Context, orgId uuid.UUID, listParams ListParams, groupBy []string, busyOnly bool) ([]map[string]any, error)
```

Comment:
As part of [EDM-1025](https://issues.redhat.com/browse/EDM-1025), we want tasks to use the service handler methods rather than access the store directly.  Can we use existing ReplaceDevice, ListDevices, etc. for these?  What's missing?

Reply:
Assuming we need it for large scale, it is preferable to work with small data sets and offload computations (i.e count) to the database.  So these operations do exactly that, so we can avoid bringing large sets of devices just for counting in order to make decisions what needs to be done.